### PR TITLE
Ouput improvements: colours & extra info

### PR DESCRIPTION
**Pull request by ezequielpereira**
* Added colours:
  * Blue for defenders and wire cards
  * Red for bombers and the bomb card
* Added extra information:
  * Number of possible defenders and bombers in play
  * Improved round counter
  * Players' roles & cards reveal at the end

### CODE REVIEW

File: pkg/engine/game.go [hunk 1]
```diff
@@ -202,19 +202,19 @@ func (g *Game) Cut(src, dst PlayerID, i int) error {
 		// nothing happens
 	}
 
-	// if we've reached the end of the round, deal a new hand
-	cuts := len(g.Cuts)
-	if cuts > 0 && cuts%n == 0 {
-		g.deal()
-		g.Round++
-	}
-
```
Comment (amlweems) [inline pkg/engine/game.go:211]: Without the g.Round++ on line 209, the else if on line 215 does not detect game over at round 4 and will continue for an extra cut into round 5.

File: pkg/tui/tui.go [hunk 2]
```diff
@@ -6,6 +6,13 @@ import (
 
 	"github.com/amlweems/timebomb/pkg/engine"
 	tea "github.com/charmbracelet/bubbletea"
+	"github.com/muesli/termenv"
+)
+
+var (
+	color = termenv.EnvColorProfile().Color
+	red = termenv.Style{}.Foreground(color("9")).Styled
+	blue = termenv.Style{}.Foreground(color("12")).Styled
```
Comment (amlweems) [inline pkg/tui/tui.go:15]: For some reason these colors are not working for me locally, do you know if I need some special terminal config?

File: pkg/engine/game.go (hunk 1 above)
Comment (ezequielpereira) [reply pkg/engine/game.go:211]: b9443a3 should fix this

File: pkg/tui/tui.go (hunk 2 above)
Comment (ezequielpereira) [reply pkg/tui/tui.go:15]: I'm not sure how it works, but when I connect locally or from my corp laptop, both running desktop Linux, I see the following entry:
```
2024/02/08 18:12:27 INFO eze connect 127.0.0.1:42388 false [cvr] xterm-256color 275 73
```

I imagine `256color` indicates that it supports these colours, maybe your computer is running something different?

But given that the text & colour is generated server-side, I'm not sure how to handle those cases.

**Pull request merged**